### PR TITLE
v3: Update more endpoints to Discord API v9

### PR DESCRIPTION
**Pull request by samhza**
Update message editing/sending for both API and webhooks.

### CODE REVIEW

**Comment by samhza** [comment]
The `embed` field for editing/sending messages has been deprecated in favor of an `embeds` field, allowing for sending/editing multiple embeds in one message. Should EditMessage/EditEmbed's API be kept the same but internally using `Embeds: &[]discord.Embed{embed}` instead of `Embed: embed`? Either way, I think we should remove the `Embed` field from {Send,Edit}MessageData.

https://discord.com/developers/docs/resources/channel#create-message

**Comment by diamondburned** [comment]
@samhza 

I agree that we should remove the singular Embed field and use the multiple one instead.

File: api/message.go [hunk 1]
```diff
@@ -230,7 +234,7 @@ func (c *Client) SendEmbed(
 	channelID discord.ChannelID, e discord.Embed) (*discord.Message, error) {
```
Comment (diamondburned) [inline api/message.go]: Would `SendEmbed(discord.ChannelID, ...discord.Embed)` be a better API for this?

File: api/message.go [hunk 2]
```diff
@@ -259,11 +263,13 @@ func (c *Client) SendEmbedReply(
 // Fires a Message Create Gateway event.
 func (c *Client) SendMessage(
 	channelID discord.ChannelID, content string, embed *discord.Embed) (*discord.Message, error) {
```
Comment (diamondburned) [inline api/message.go]: I think this can also do `SendMessage(discord.ChannelID, string, ...discord.Embed)`.

File: api/message.go [hunk 3]
```diff
@@ -318,7 +339,7 @@ func (c *Client) EditEmbed(
 	channelID discord.ChannelID, messageID discord.MessageID, embed discord.Embed) (*discord.Message, error) {
```
Comment (diamondburned) [inline api/message.go]: (above 2 comments)

File: utils/sendpart/sendpart.go [hunk 4]
```diff
@@ -37,16 +38,15 @@ type DataMultipartWriter interface {
 	httputil.MultipartWriter
 }
 
-// POST sends a POST request using client to the given URL and unmarshal the
-// body into v if it's not nil. It will only send using multipart if files is
-// true.
-func POST(c *httputil.Client, data DataMultipartWriter, v interface{}, url string) error {
+// Do sends an HTTP request using client to the given URL and unmarshals the
+// body into v if it's not nil. It will only send using multipart if needed.
+func Do(method string, c *httputil.Client, data DataMultipartWriter, v interface{}, url string) error {
```
Comment (diamondburned) [inline utils/sendpart/sendpart.go]: The order of these arguments should be `Do(c *httputil.Client, method string, data DataMultipartWriter, v interface{}, url string) error`.

File: utils/sendpart/sendpart.go [hunk 5]
```diff
@@ -61,6 +61,20 @@ func POST(c *httputil.Client, data DataMultipartWriter, v interface{}, url strin
 	return json.DecodeStream(body, v)
 }
 
+// PATCH sends a PATCH request using client to the given URL and unmarshals the
+// body into v if it's not nil. It will only send using multipart if needed.
+// It is equivalent to calling Do with http.MethodPost
+func POST(c *httputil.Client, data DataMultipartWriter, v interface{}, url string) error {
+	return Do(http.MethodPost, c, data, v, url)
```
Comment (diamondburned) [inline utils/sendpart/sendpart.go]: The `http.Method*` constants probably shouldn't be used here to avoid importing `net/http` just for those constants.

**Review by diamondburned — CHANGES_REQUESTED** [review]

**Review by diamondburned — APPROVED** [review]

**Pull request merged**